src/sentry/utils/silo/common.py

```diff
@@ -41,24 +61,40 @@ def insert_import(src_code: str) -> str:
         else:
             return import_stmt + "\n" + src_code
 
-    def is_module(src_path: str, module_name: str | None) -> bool:
-        if module_name is None:
-            return False
-        suffix = ".py"
-        return src_path.endswith(suffix) and (
-            src_path[: -len(suffix)].replace("/", ".").endswith(module_name)
+    for src_path, class_name in find_class_declarations():
+        with open(src_path) as f:
+            src_code = f.read()
+        decorator_is_new = False
+
+        def replace(match):
+            nonlocal decorator_is_new
+
+            existing_decorator, class_decl = match.groups()
+            if existing_decorator == decorator_name:
+                return match.group()
+            decorator_is_new = True
+            return f"\n@{decorator_name}\n{class_decl}"
+
+        # Try to detect a pre-existing decorator, if possible, and replace it if it
+        # doesn't match the decorator we are trying to add. This works only if the
+        # decorator is immediately above the class, which is usually a safe
+        # assumption because that's where this script writes it. In the long run,
+        # it could be disrupted by the insertion of a comment, another unrelated
+        # decorator, or such.
+        new_code = re.sub(
+            (
+                r"(?:\n@((?:control|customer|pending)_silo_(?:model|endpoint|test)))?"
+                rf"\n(class\s+{class_name}\()"
+            ),
+            replace,
+            src_code,
```

Comment:
Looking for honest feedback -- is this too much nutty regex magic? I wouldn't be okay with letting this level of hairiness into long-term production code. But seeing as these scripts are meant to have a limited lifespan, I think I'm okay with favoring power over clarity.

Also, previously `add_silo_decorators` was the sole authority (in the `utils.silo` package) on the set of decorator names. I don't feel great about spreading that information here (with the `(?:control|customer|pending)_silo_(?:model|endpoint|test)` bit) but 🤷 it's probably fine.

Reply:
Mark a TODO, and file a issue for us to deal with before GA.

Reply:
I think this level of regex is totally reasonable for a code generation tool. The alternative would be to use an AST parser and taking on the complexity of doing AST transforms.

Reply:
By the time this goes to GA, we should be able to delete this script package entirely.